src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -1142,6 +1151,10 @@ void TCPTransportInterface::SocketConnected(Locator_t locator, const asio::error
         }
         else
         {
+            if (error.value() == 111)
+            {
+                eClock::my_sleep(200); // Wait a little before try again.
```

Comment:
Also explain the reason for this sleep.

Reply:
Explained in the next commit.

Reply:
If I understand correctly, this is to prevent an infinite loop of connect->refuse->connect->refuse calls.

Reply:
Exactly, once the OS exhausts all out possible file descriptors it will fail forever.